sda-download/go.mod

```
@@ -1,6 +1,6 @@
 module github.com/neicnordic/sda-download
 
-go 1.21
+go 1.22.2
```

i think we don\t need the patch version ...
```suggestion
go 1.22
```